File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> answer;
    
    auto it = min_element(arr.begin(), arr.end());
    cout << *it << endl;
    
    if(arr.size() != 1)
    {
        arr.erase(it);
        answer = arr;
    }
    else
        answer.push_back(-1);
    
    return answer;
}

File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.cs
using System;
using System.Linq;

public class Solution {
    public int[] solution(int[] arr) {
        int[] answer = new int[] {};
        
        if(arr.Length == 1)
        {
            Array.Resize(ref answer, 1);
            answer[0] = -1;
        }
        else
        {
            Array.Resize(ref answer, arr.Length - 1);
            
            int minVal = arr.Min();
            int idx = 0;
            for(int i = 0; i < arr.Length; ++i)
            {
                if(arr[i] != minVal)
                {
                    answer[idx] = arr[i];
                    ++idx;
                }
            }
        }
        
        return answer;
    }
}